Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -275,6 +293,29 @@ class AttachmentSaveEventListener final : public EventListener {
     WeakPtr<HTMLAttachmentElement, WeakPtrImplWithEventTargetData> m_attachment;
 };
 
+void HTMLAttachmentElement::updateProgress(const AtomString& progress)
+{
+    if (!m_progressElement)
+        return;
+
+    bool validProgress = false;
+    float value = progress.toFloat(&validProgress);
+    if (validProgress && std::isfinite(value)) {
+        StringBuilder style;
+        style.append("--progress: "_s);
+        if (value < 0.0)
+            style.append("0"_s);
+        else if (value > 1.0)
+            style.append("1"_s);
```

Comment:
Can we use `std::clamp` to normalize the value, and avoid these conditionals?

Reply:
I wanted to avoid doing a float->string conversion when it can be bypassed here (since we already have the string in most cases).
Would you still prefer clamping + to-string'ing the value?

Reply:
We could clamp and only convert to string if the values are not equal.

I don't feel too strongly about this though.

Reply:
> only convert to string if the values are not equal
Possible, but I'm afraid it would become more complex code!
Thanks for your understanding.